### ISSUE TITLE
AVC low power encoding fails on ubuntu 19.04

### ISSUE THREAD

**Issue by dvrogozh** [Intel-Media-SDK/MediaSDK#1398]
Install Ubuntu 19.04 and mediasdk coming from the OS:
```
sudo apt-get install libmfx1 libmfx-tools
sudo apt-get install libva-drm2 libva-x11-2
```

And run the following sample encode command line to encode via AVC Low Power encoder:
```
export LIBVA_DRIVER_NAME=iHD

# just to prepare some YUV input file:
wget https://fate-suite.libav.org/h264-conformance/AUD_MW_E.264
/usr/share/mfx/samples/sample_decode h264 -i AUD_MW_E.264 -o AUD_MW_E.yuv

# and now encode:
sample_encode: /build/intel-mediasdk-AzMSof/intel-mediasdk-18.4.1/_studio/mfx_lib/shared/src/mfx_h264_encode_vaapi.cpp:1422: virtual mfxStatus MfxHwH264Encode::VAAPIEncoder::CreateAuxilliaryDevice(VideoCORE*, GUID, mfxU32, mfxU32, bool): Assertion `0x00000000 == vaSts' failed.

Aborted (core dumped)
```

Please, make sure to have `intel-media-va-driver` installed instead of `intel-media-va-driver-non-free` since the latter on does not return the error on the above command line:
```
sudo apt-get install intel-media-va-driver
```

**Comment by dvrogozh** [comment]
There are few issues here:
1. gcc-8.3.0 compiler on Ubuntu 19.04 aligns mediasdk internally defined GUID structure size by 8 and we compare non-initialized memory via memcmp inside operator== for GUIDs. As a result we fail to recognize that user requested LowPower encoding and try to initialize Dual Pipe which is not available on `intel-media-va-driver`
2. I suspect that in case of the full `intel-media-va-driver-non-free` we actually run Dual Pipe encoding instead of Low Power for the same reason as explained above
3. PR #1399 fixes the issue in mediasdk to compare GUIDs correctly (but it does nothing to assure that user will get Low Power if he requested it explicitly). However, there is a *second issue in media-driver* because of which Low Power encoding still won't work after #1399. Effectively, https://github.com/intel/media-driver/releases/tag/intel-media-18.4.1 release miss the following patch: https://github.com/intel/media-driver/commit/26d7d503c993f46377f6366ac076e6fc98ec179f

**Comment by dvrogozh** [comment]
Filed https://github.com/intel/media-driver/issues/613 for driver part.

**Comment by dvrogozh** [comment]
Fixed w/ mediasdk package update in Ubuntu 19.10. Please, consider updating Ubuntu regular release from 19.04 to 19.10 if you are affected by this issue.
